### ISSUE TITLE
Map.containsKey(...) logic support for string constants

### ISSUE THREAD

**Issue by lazaroclapp** [uber/NullAway#398]
This works:

```
String x = "constant";
if (m.containsKey(x)) {
  Object y = m.get(x);
  y.toString();
}
```

This doesn't:

```
if (m.containsKey("constant")) {
  Object y = m.get("constant");
  y.toString(); // dereference error here
}
```

Both probably should.

Haven't thought about how hard the second case might be, but the difference can be really weird from a developer/user point of view :)

**Comment by msridhar** [comment]
For the second case the hackish thing would be to have some kind of string constant "access path."  To do things more properly I guess we could introduce an interface for "things that can be an argument to `Map` methods" and then have both `AccessPath` and some string constant type implement the interface.

**Comment by lazaroclapp** [comment]
Closed by #413
